pkg/controller/autoscaling/elasticsearch/autoscaler/linear_scaler.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package autoscaler
+
+import (
+	"fmt"
+
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// cpuFromMemory computes a CPU quantity within the specified allowed range by the user proportionally
+// to the amount of memory requested by the autoscaling API.
+func cpuFromMemory(requiredMemoryCapacity resource.Quantity, memoryRange, cpuRange esv1.QuantityRange) resource.Quantity {
+	allowedMemoryRange := memoryRange.Max.Value() - memoryRange.Min.Value()
+	if allowedMemoryRange == 0 {
+		// Can't scale CPU as min and max for memory are equal
+		return cpuRange.Min.DeepCopy()
+	}
+	memRatio := float64(requiredMemoryCapacity.Value()-memoryRange.Min.Value()) / float64(allowedMemoryRange)
+
+	// memory is at its lowest value, return the min value for CPU
+	if memRatio == 0 {
+		return cpuRange.Min.DeepCopy()
+	}
+	// memory is at its max value, return the max value for CPU
+	if memRatio == 1 {
+		return cpuRange.Max.DeepCopy()
+	}
+
+	allowedCPURange := float64(cpuRange.Max.MilliValue() - cpuRange.Min.MilliValue())
+	requiredAdditionalCPUCapacity := int64(allowedCPURange * memRatio)
+	requiredCPUCapacityAsMilli := cpuRange.Min.MilliValue() + requiredAdditionalCPUCapacity
+
+	// Round up memory to the next core
+	requiredCPUCapacityAsMilli = roundUp(requiredCPUCapacityAsMilli, 1000)
```

Comment:
Should we allow CPU capacity in smaller increments than one core?

Reply:
I don't have a strong opinion. I always have preferred rounded values because it makes it easier to resonate about the performance model (e.g. how to size thread pools in the JVM with fractional number ?) But I agree that the user might want to have more accurate resource requests.
@elastic/cloud-k8s I'm curious if there are other opinions.

Reply:
For Elasticsearch or Kibana, we would round up CPU capacity to an integer, but might be too much for lighter components, for example beats. 
Also, if you just want to run the whole ECK stack on a laptop, users would prefer to have less granual option to define resource limits, just to enable them to test locally.